### DISCUSSION TITLE
Bundled/exported components render slowly locally but normally in production. Ideas why?

### DISCUSSION THREAD

**Discussion by brandon-selfbook** [styled-components/styled-components#4221] [Q&A]
I have a component-library where my `NumberInput` component renders fully in ~11ms. When I build the library and use that exported component in an app locally it takes upwards of ~680ms to load. When _that_ app is built for production the component again loads in less than ~50ms.

Anyone have thoughts on why this is happening? Clicking on UI that results in this component rendering locks up the app and mouse until it renders. Could this be an issue with my build tool (Rollup)?

**Comment by brandon-selfbook** [comment]
Closing as the issue was actually with base64 encoded fonts.

**Comment by quantizor** [comment]
This is very likely caused by `DISABLE_SPEEDY` — in development mode, styled-components uses DOM text node injection instead of the CSSOM API, which is significantly slower in dev environments. In production builds, the fast CSSOM path (`CSSStyleSheet.insertRule`) is used, explaining the dramatic difference.

The upcoming release includes performance improvements in this area.

You can test the prerelease:

```
npm install styled-components@test
```

Staged release PR: https://github.com/styled-components/styled-components/pull/5687
